third_party/txt/src/txt/paragraph.cc

```diff
@@ -440,7 +437,25 @@ void Paragraph::Layout(double width, bool force) {
               current_x_position + previous_run_x_position + letter_spacing);
           buffers.back()->pos[pos_index + 1] = layout.getY(glyph_index);
 
-          current_x_position += layout.getCharAdvance(glyph_index);
+          float glyph_advance = layout.getCharAdvance(character_index);
+          current_x_position += glyph_advance;
+
+          // The glyph may be a ligature.  Determine how many input characters
+          // are joined into this glyph.
+          size_t ligature_end = character_index + 1;
+          for (; ligature_end < text_count; ++ligature_end) {
+            if (layout.getCharAdvance(ligature_end) != 0)
+              break;
+          }
+          size_t ligature_width = ligature_end - character_index;
+          prev_char_advance = glyph_advance / ligature_width;
+          // Compute positions for the additional characters in the ligature.
+          for (size_t i = 1; i < ligature_width; i++) {
```

Comment:
`++i` for consistency... not that the compiler won't optimise that away anyway :-)